File: easy/palindrome_number.rb
# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
  str = x.to_s
  half_legnth = str.length / 2
  if str[0..(half_legnth-1)] == str[(-half_legnth)..(-1)].reverse
    return true
  else
    return false
  end
end

File: easy/palindrome_number.py
class Solution(object):
  def isPalindrome(self, x):
    """
    :type x: int
    :rtype: bool
    """
    str_num = str(x)
    full_length = len(str_num)
    half_legnth = len(str_num) / 2
    if str_num[:(half_legnth)] == str_num[(full_length-half_legnth):][::-1]:
      return True
    else:
      return False
